BHoM_UI/Components/oM/CreateObject.cs

```diff
@@ -53,9 +57,157 @@ public class CreateObjectCaller : MethodCaller
+            else if (SelectedItem is Type)
+            {
+                if (m_CompiledFunc == null)
+                    m_CompiledFunc = CreateConstructor((Type)SelectedItem, InputParams);
+                return m_CompiledFunc(inputs);
+            }
+            else
+                return null;
         }
 
-            /*************************************/
+        /*************************************/
+
+        public override bool SetItem(object item)
+        {
+            if (item is MethodBase)
+            {
+                return base.SetItem(item);
+            }
+            else if (item is Type)
+            {
+                SelectedItem = item;
+                Type type = item as Type;
+                Name = type.Name;
+                Description = type.Description();
+
+                //object instance = Activator.CreateInstance(type);  // Potentially for later
+                string[] excluded = new string[] { "BHoM_Guid" };
+                InputParams = type.GetProperties().Where(x => !excluded.Contains(x.Name)).Select(x => x.ToBHoM()).ToList();
+
+                OutputParams = new List<ParamInfo>() { new ParamInfo { DataType = type, Kind = ParamKind.Output, Name = Name.Substring(0, 1), Description = type.Description() } };
+                m_CompiledFunc = CreateConstructor(type, InputParams);
+
+                CompileInputGetters();
+                CompileOutputSetters();
+            }
+            return true;
+        }
+
+        /*************************************/
+
+        public override bool AddInput(int index, string name, Type type = null)
+        {
+            if (SelectedItem is Type)
+            {
+                PropertyInfo info = ((Type)SelectedItem).GetProperty(name);
+                if (info != null)
+                    type = info.PropertyType;
+            }
+
+            return base.AddInput(index, name, type);
         }
+
+        /*************************************/
+
+        public override bool RemoveInput(string name)
+        {
+            if (name == null)
+                return false;
+
+            bool success = InputParams.RemoveAll(p => p.Name == name) > 0;
+            CompileInputGetters();
+
+            if (SelectedItem is Type)
+                m_CompiledFunc = CreateConstructor((Type)SelectedItem, InputParams);
+            return success;
+        }
+
+        /*************************************/
+
+        public override bool Read(string json)
+        {
+            if (!base.Read(json))
+                return false;
+
+            if (SelectedItem is Type)
+                m_CompiledFunc = CreateConstructor((Type)SelectedItem, InputParams);
+
+            return true;
+        }
+
+
+        /*************************************/
+        /**** Private Methods             ****/
+        /*************************************/
+
+        protected Func<object[], object> CreateConstructor(Type type, List<ParamInfo> parameters)
```

Comment:
In the light of the migration of all the methods to construct delegates (https://github.com/BHoM/BHoM_Engine/blob/master/Reflection_Engine/Convert/ToFunc.cs) maybe `BH.Engine.Reflection.Create` is a better place for this?

Reply:
Good point